src/mochiweb_http.erl

```diff
@@ -168,8 +174,14 @@ collect_headers(Socket, Request, Body, Collected, Trunc, HeaderCount) ->
         {ssl_closed, _} ->
             mochiweb_socket:close(Socket),
             exit(normal);
-        _Other ->
-            handle_invalid_request(Socket, Request, [])
+        {ProtocolErr, _Socket, _Reason} when ProtocolErr =:= tcp_error orelse ProtocolErr =:= ssl_error ->
```

Comment:
If you were diagnosing a production application, how would you look for this occurrence?

It might be nicer to add a ```close_on_error(Socket, _Reason)``` function with an ignored reason so somebody could trace for it without having to trace all messages to mochiweb processes.

That does mean changing more of the original code, but does give us more of a hope debugging in the field.